daemon/volumes.go

```diff
@@ -184,6 +184,12 @@ func (container *Container) parseVolumeMountConfig() (map[string]*Mount, error)
 			continue
 		}
 
+		if stat, err := os.Stat(filepath.Join(container.basefs, path)); err == nil {
```

Comment:
@cpuguy83 should this be Lstat? What happens if what would become the volume path, is a symlink to a folder?Should it follow the symlink and then "merge" with the folder? Or should we error out because a symlink is just a file? In case we error out because we consider the symlink being a file, then we should add a test.

I think it makes sense to follow the symlink, but I'd like to hear others' opinion /cc @tianon @jfrazelle